### PR TITLE
Port #50135 to master

### PR DESCRIPTION
**Pull request by terminalmage**
See https://github.com/saltstack/salt/pull/50135

### CODE REVIEW

**Comment by terminalmage** [comment]
Test case added.

File: salt/cloud/clouds/saltify.py [hunk 1]
```diff
@@ -268,6 +268,10 @@ def create(vm_):
         "deploy", vm_, __opts__, default=False
     )
 
+    # If ssh_host is not set, default to the minion name
+    if not config.get_cloud_config_value("ssh_host", vm_, __opts__, default=""):
+        vm_["ssh_host"] = vm_["name"]
```
Comment (DmitryKuzmenko) [inline salt/cloud/clouds/saltify.py:273]: Why not to provide the `vm_["name"]` as a `default` value to the `get_cloud_config_value` call?

Comment (terminalmage) [reply salt/cloud/clouds/saltify.py:273]: I didn't write the original PR, but I think the reason is that this function call is just a lookup to see if `ssh_host` was set in the cloud config. If not, then the key gets set on line 273. So if you set `vm_["name"]` as the default on line 272, then we will never reach line 273, and `vm_["ssh_host"]` will never be set.

Comment (terminalmage) [reply salt/cloud/clouds/saltify.py:273]: No problem!

Comment (terminalmage) [reply salt/cloud/clouds/saltify.py:273]: Should be good to merge then, unless this flaky codecov check matters.

**Review by dwoz — APPROVED** [review]

**Pull request merged**